narwhals/_duckdb/namespace.py

```diff
@@ -157,6 +158,16 @@ def func(df: DuckDBLazyFrame) -> list[duckdb.Expression]:
             kwargs={"exprs": exprs},
         )
 
+    def when(
+        self,
+        *predicates: IntoDuckDBExpr,
+    ) -> DuckDBWhen:
+        plx = self.__class__(backend_version=self._backend_version, version=self._version)
+        condition = plx.all_horizontal(*predicates)
```

Comment:
```suggestion
        if predicates:
            condition = plx.all_horizontal(*predicates)
        else:
            msg = "at least one predicate needs to be provided"
            raise TypeError(msg)
```

The other backends have this check

Reply:
ah i moved it up cause i was tired of rewriting it everywhere 😄  https://github.com/narwhals-dev/narwhals/pull/1756

Reply:
nice!